ydb/docs/en/core/concepts/optimizer.md

```diff
@@ -0,0 +1,81 @@
+
+## Cost-Based Query Optimizer
+
+Cost-Based Optimizer is used for more complex optimizations, such as choosing an optimal join order and join algorithms. Cost-based Optimizer considers a large number of alternative execution plans for each query and chooses the best one based on the cost estimate for each option. Currently, this optimizer only works with plans that contain [JOIN](../yql/reference/syntax/join.md) operations. It chooses the best order for these operations to be executed, and also chooses the most efficient implementation for each Join algoritm in the plan.
+
+The cost optimizer consists of three main components:
+
+* Plan enumerator
+* Cost estimation function
+* Statistics module, which is used to estimate statistics for the cost function
+
+### Plan enumerator
+
+The current Cost-Based Optimizer in {{ ydb-short-name }} enumerates all useful join trees, for which the join conditions are defined. We first build a join hypergraph, where the nodes are tables and edges are join conditions. Depending on how the original query was written, the join hypergraph may have quite different topologies, varying from simple chain-like graphs to complex cliques. The resulting topology of the join graph determines how many possible alternite plans need to be considered by the optimizer.
+
+For example, a star is a common topology in analytical queries, where a main fact table is joined to multiple dimension tables:
+
+```sql
+SELECT
+    P.Brand,
+    S.Country AS Countries,
+    SUM(F.Units_Sold)
+
+FROM Fact_Sales F
+INNER JOIN Dim_Date D    ON (F.Date_Id = D.Id)
+INNER JOIN Dim_Store S   ON (F.Store_Id = S.Id)
+INNER JOIN Dim_Product P ON (F.Product_Id = P.Id)
+
+WHERE D.Year = 1997 AND  P.Product_Category = 'tv'
+
+GROUP BY
+    P.Brand,
+    S.Country
+```
+
+In this query graph, all `Dim...` tables are joined to the `Fact_Sales` fact table:
+![Join Graph](_assets/Star-Schema.png)
+
+Common topologies also include chain and clique. A "chain" is a topology where tables are connected to each other sequentially and each table participates in no more than one join. A "clique" is a fully connected graph where each table is connected to another.
+
+In practice, OLAP queries often have a topology that is a combination of a "star" and a "chain", while complex topologies like a "clique" are very rare.
+
+The topology greatly influences the number of alternative plans that the optimizer needs to consider. Therefore, the cost-based optimizer limits the number of joins that are compared by exhaustive search, depending on the topology of the original plan. The capabilities of exact optimization in {{ ydb-short-name }} are listed in the following table:
```

Comment:
```suggestion
Common topologies also include chains and cliques. A "chain" is a topology where tables are connected sequentially, with each table participating in no more than one join. A "clique" is a fully connected graph where each table is connected to every other table.

In practice, OLAP queries often exhibit a combination of "star" and "chain" topologies, while complex topologies like a "clique" are very rare.

The topology significantly impacts the number of alternative plans the optimizer must consider. Therefore, the cost-based optimizer limits the number of joins that are evaluated through an exhaustive search, depending on the topology of the original plan. The capabilities of exact optimization in {{ ydb-short-name }} are listed in the following table:
```

Reply:
I don't agree to these changes - especially 'evaluated though an' - this is more confusing, it gives an impression that we are evaluating joins - which is not the case